airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -486,6 +488,12 @@ def get_deployment_status(
         except Exception as exc:
             raise exc
 
+    @tenacity.retry(
```

Comment:
This will be retried for non transient errors as well.  can't we rely on the task retry instead of explicit retry here?

Reply:
Yeah - it's more "resilience" than actual retry.